RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -15,91 +16,94 @@
-                private long currentSize = 0;
-
-                @Override
-                public boolean test(BulkDocSection next) {
-                    // TODO: Move to Bytebufs to convert from string to bytes only once
-                    // Add one for newline between bulk sections
-                    var nextSize = next.asBulkIndex().length() + 1L;
-                    currentSize += nextSize;
-                    currentItemCount++;
-
-                    if (currentItemCount > maxDocsPerBulkRequest || currentSize > maxBytesPerBulkRequest) {
-                        // Reset and return true to signal to stop buffering.
-                        // Current item is included in the current buffer
-                        currentItemCount = 1;
-                        currentSize = nextSize;
-                        return true;
-                    }
-                    return false;
-                }
-            }, true)
-             .parallel(
-                maxConcurrentWorkItems, // Number of parallel workers, tested in reindex_shouldRespectMaxConcurrentRequests
-                maxConcurrentWorkItems // Limit prefetch for memory pressure
-            )
-            .flatMap(
-                bulkDocs -> client
-                    .sendBulkRequest(indexName, bulkDocs, context.createBulkRequest()) // Send the request
-                    .doFirst(() -> log.atInfo().log("{} documents in current bulk request.", bulkDocs.size()))
-                    .doOnSuccess(unused -> log.atDebug().log("Batch succeeded"))
-                    .doOnError(error -> log.atError().log("Batch failed", error))
-                    // Prevent the error from stopping the entire stream, retries occurring within sendBulkRequest
-                    .onErrorResume(e -> Mono.empty()),
-            false,
-                1, // control concurrency on parallel rails
-                1 // control prefetch across all parallel runners
-            )
-            .doOnComplete(() -> log.debug("All batches processed"))
+    public Mono<Void> reindex(String indexName, Flux<Document> documentStream, IDocumentReindexContext context) {
+        // Create scheduler for short-lived CPU bound tasks
+        var scheduler = Schedulers.newParallel("DocumentReader");
+        var bulkDocs = documentStream.publishOn(scheduler).map(BulkDocSection::new);
+
+        return this.reindexDocsInParallelBatches(bulkDocs, indexName, context)
+            .doOnSuccess(unused -> log.debug("All batches processed"))
+            .doOnError(e -> log.error("Error prevented all batches from being processed", e))
+            .doOnTerminate(scheduler::dispose);
+    }
+
+    Mono<Void> reindexDocsInParallelBatches(Flux<BulkDocSection> docs, String indexName, IDocumentReindexContext context) {
+        // Create elastic scheduler for long-lived i/o bound tasks
+        var scheduler = Schedulers.newBoundedElastic(maxConcurrentWorkItems, Integer.MAX_VALUE, "DocumentBatchReindexer");
```

Comment:
How does the new scheduler impact the execution flow and performance?  I'm guessing there was a default scheduler we were implicitly using originally.